core/src/main/scala/neotypes/Driver.scala

```diff
@@ -8,14 +8,42 @@ import org.neo4j.driver.v1.{AccessMode, Driver => NDriver}
 import scala.jdk.CollectionConverters._
 import scala.language.higherKinds
 
+/** neotypes driver for accessing a neo4j graph database
+  *
+  * @param driver neo4j driver
+  * @tparam F effect type for driver
+  */
 final class Driver[F[_]](private val driver: NDriver) extends AnyVal {
+
+  /** acquire session to a database in read mode
+    * @note a session is a resource that needs to be allocated and released
+    *
+    * @param F asynchronous effect type with resource type defined
+    * @tparam R effect type that allows creating, using, and releasing a resource
+    * @return Session[F] in effect type R
```

Comment:
`R` is not really an **effect type** it is rather a **resource type** which is dependent of the `F` **effect type** which was used for the creation of the Driver.

Also, we should be very careful when talking about this as since for **Future** the _"resource"_ type is the **Id** meaning there isn't really a resource.

Reply:
Yeah, I understand your concern here. I'd say *resource type* works for me and maybe we can add a note that would emphasize on this exception with *Future*

Reply:
Yeah seems like the best way.

Reply:
I bit tedious... But, it would be worth adding the same note about resources and future here.

But then, it would be good adding that note on all the methods that return a resource, and there are many.
It would be good if we could have the message in one place and reference it in many, as a variable, so we do not have to repeat ourselves and making changes in a future would be easier.

I am pretty sure that exists. I believe to remember seeing something like that on the official scaladoc, precisely on the collections framework.
Could you help us looking for that? It would be really helpful!!!

Reply:
@geoffjohn11 Pretty cool!! Thank.

But now the thing is, that note will be also used in the **Session** class and in the **GraphDatabase** object.
You know if there is a way to have that `@define` in a global space? _(or it is already global)_. Maybe we could put it in another file, like `globals_doc.scala`?